### PR TITLE
General code cleanup [full build]

### PR DESCRIPTION
**Pull request by porunov**
General code cleanup by Intellij IDEA

This PR makes next changes:
- Removes access modifiers from interfaces
- Removes unnecessary transformation to strings strings concatenations
- Removes redundant `throws` when higher level exception exists
- Removes redundant `throws` when the Exception isn't thrown
- Removes `length` from substring when using with `str.substring(from, str.length())` (redundant)
- Changes arrays declaration from old style to normal style. I.e. from `Type variableName[]` to `Type[] variableName`
- Changes `.class.isInstance(class)` to direct `instanceof` operator
- `Charset.forName("UTF-8")` to `StandardCharsets.UTF_8`
- Some  ternary operators to boolean operators if possible
-  Redundant casting to some types
- Redundant semicolons from Groovy files
- Redundant `def` from Groovy files
- Redundant `RuntimeException` wrapping
- Adds figure braces to all `if` statements in groovy files

Signed-off-by: Oleksandr Porunov <alexandr.porunov@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-core/src/main/java/org/janusgraph/core/schema/JanusGraphManagement.java [hunk 1]
```diff
@@ -410,34 +410,34 @@
      *
      * @return a collection of all the mgmt API schema printing methods
      */
-    public String printSchema();
+    String printSchema();
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/schema/JanusGraphManagement.java]: Doesn't this change make these methods package-private?

File: janusgraph-test/src/main/java/org/janusgraph/diskstorage/KeyColumnValueStoreUtil.java [hunk 2]
```diff
@@ -86,22 +87,14 @@ public static void loadValues(KeyColumnValueStore store, StoreTransaction tx, St
...
 
     // TODO rename as "stringToBuffer" after syntax errors are resolved
     public static StaticBuffer stringToByteBuffer(String s) {
         byte[] b;
-        try {
-            b = s.getBytes("UTF-8");
-        } catch (UnsupportedEncodingException e) {
-            throw new RuntimeException(e);
-        }
+        b = s.getBytes(StandardCharsets.UTF_8);
```
Comment (farodin91) [inline janusgraph-test/src/main/java/org/janusgraph/diskstorage/KeyColumnValueStoreUtil.java]: merge line 96 and 97.

File: janusgraph-hadoop-parent/janusgraph-hadoop-core/src/test/resources/org/janusgraph/hadoop/formats/graphson/incremental-naive-cerberus-load.groovy [hunk 3]
```diff
@@ -12,20 +12,20 @@
...
-    Iterator<Vertex> itty = graph.query().has(uniqueKey, uniqueValue).vertices().iterator();
+      throw new RuntimeException("The provided Faunus vertex does not have a property for the unique key: " + faunusVertex)
+
+    Iterator<Vertex> itty = graph.query().has(uniqueKey, uniqueValue).vertices().iterator()
     if (itty.hasNext()) {
-      janusgraphVertex = itty.next();
-      if (itty.hasNext())
-        log.info("The unique key is not unique as more than one vertex with the value {}", uniqueValue);
+      janusgraphVertex = itty.next()
+        if (itty.hasNext())
```
Comment (FlorianHockmann) [inline janusgraph-hadoop-parent/janusgraph-hadoop-core/src/test/resources/org/janusgraph/hadoop/formats/graphson/incremental-naive-cerberus-load.groovy]: (nitpick) This indent looks wrong to me. Should the `if` have the same indentation as the line above?

File: janusgraph-core/src/main/java/org/janusgraph/core/schema/JanusGraphManagement.java (hunk 1 above)
Comment (farodin91) [reply janusgraph-core/src/main/java/org/janusgraph/core/schema/JanusGraphManagement.java]: No, this file is an interface. https://stackoverflow.com/a/161787

Comment (porunov) [reply janusgraph-core/src/main/java/org/janusgraph/core/schema/JanusGraphManagement.java]: As said by @farodin91 . All methods in the interface are `public`. This is redundant

File: janusgraph-hadoop-parent/janusgraph-hadoop-core/src/test/resources/org/janusgraph/hadoop/formats/graphson/incremental-naive-cerberus-load.groovy (hunk 3 above)
Comment (porunov) [reply janusgraph-hadoop-parent/janusgraph-hadoop-core/src/test/resources/org/janusgraph/hadoop/formats/graphson/incremental-naive-cerberus-load.groovy]: Hm. Looks like it should have the same indentation level. I don't know why Intelij IDEA made this. I will fix it

Comment (porunov) [reply janusgraph-hadoop-parent/janusgraph-hadoop-core/src/test/resources/org/janusgraph/hadoop/formats/graphson/incremental-naive-cerberus-load.groovy]: Fixed here: https://github.com/JanusGraph/janusgraph/compare/d8f481001970915363ff38a52517ee3ae963d0d1..073c7512102cdd5de723c0740406ec705b9fae6a

Comment (porunov) [reply janusgraph-hadoop-parent/janusgraph-hadoop-core/src/test/resources/org/janusgraph/hadoop/formats/graphson/incremental-naive-cerberus-load.groovy]: Fixed other similar problems here: https://github.com/JanusGraph/janusgraph/compare/073c7512102cdd5de723c0740406ec705b9fae6a..f69337316186edf248f64b792dc5258bc9fbd8f5

**Comment by porunov** [comment]
@FlorianHockmann I've described changes by this PR in the PR description

**Comment by porunov** [comment]
@FlorianHockmann @farodin91 I have changed the branch to `0.3`

**Comment by porunov** [comment]
I have merged changes to forked repository and waiting for tests to be finished. When tests are finished for this PR + forked branches `0.3`, `0.4` and `master` then I will push merge commits to the JanusGraph repository.

Changes for `0.3` branch:
https://github.com/porunov/janusgraph/commit/e77f2d923168338024c1f6f26152120672ad73e2
`0.3` tests:
https://travis-ci.org/porunov/janusgraph/builds/572690043

Changes for `0.4` branch:
https://github.com/porunov/janusgraph/commit/0181f09bfdab98065a2e3104a24c510e09f1efbf
`0.4` tests:
https://travis-ci.org/porunov/janusgraph/builds/572690123

Changes for `master` branch:
https://github.com/porunov/janusgraph/commit/e87a5eee8b835c9ba0de2ea15d4e800c0bc12ea6
`master` tests:
https://travis-ci.org/porunov/janusgraph/builds/572689996

**Comment by porunov** [comment]
I have stop the build while `0.3` branch is failing. When it is fixed, I can rebase the branch, make new merge commits and start the testing again.

**Comment by porunov** [comment]
Commits with test containers are reverted. I will rebase this branch and start full test of all branches again

**Comment by farodin91** [comment]
@porunov We can ignore these full build until they do anything more than using the default cassandra version.

**Comment by porunov** [comment]
I have merged changes to forked repository and waiting for tests to be finished. When tests are finished for this PR + forked branches `0.3`, `0.4` and `master` then I will push merge commits to the JanusGraph repository.

Changes for `0.3` branch:
https://github.com/porunov/janusgraph/commit/14a9c734fbbb6a81fb5b2bafaab137f3fe2ecb7b
`0.3` tests:
https://travis-ci.org/porunov/janusgraph/builds/573404127

Changes for `0.4` branch:
https://github.com/porunov/janusgraph/commit/cc2b3de63b072254f413ec433bcb6479c59474e7
`0.4` tests:
https://travis-ci.org/porunov/janusgraph/builds/573403990

Changes for `master` branch:
https://github.com/porunov/janusgraph/commit/3b71cd389cd1e9f04ea389962166eeb851683f8e
`master` tests:
https://travis-ci.org/porunov/janusgraph/builds/573404056

**Comment by porunov** [comment]
@farodin91 What do you mean by ignoring it? `[full build]` is redundant then? I thought it was created to run test against all supported cassandra versions. Isn't it?

**Comment by farodin91** [comment]
@porunov You have to start before a casandra using docker-compose see line https://github.com/JanusGraph/janusgraph/blob/master/.travis.yml.cassandra#L66 and than start cassandra tests with following command https://github.com/JanusGraph/janusgraph/blob/master/.travis.yml.cassandra#L79.Otherwise the default version is used.

**Comment by porunov** [comment]
@farodin91 will you be able to make this feature (or bug fix)? 

I don't think we can run all tests on Travis right now due to the hard limit of 50 minutes per build.
I have contacted Travis sales and will have a discussion on Monday about possibility to buy Travis plans (to be able to run all test) for JanusGraph for a lower price then listed officially. The main point is that we are an open source project but we need commercial plans because they have a critical feature (long running jobs).
If they allow us to have a Premium plan for a lower price I will tell it here:  https://groups.google.com/forum/#!topic/janusgraph-dev/UpfwQD7O2XQ

We already have a small discussion there with @mbrukman but it isn't clear how can we get an infrastructure support which is needed for JanusGraph. Currently possible solutions are:
1) Ask The Linux Foundation if they can provide us with Travis Premium plans.
2) Try to raise funds by owrself to be able to afford those infrastructure.

After we have long running Travis jobs (or any other CI), I think we can enable all tests to run periodically. Before that, we most likely will have some problems with some tests which sometimes run more than 50 minutes. It includes TinkerPop tests and possibly some CQL tests

**Comment by farodin91** [comment]
It will take some time to fix it. So, I would prefer to stop run all of these full build for now. The cql tests now works but take nearly the up all the 50 minutes.

We can do some more split up and i think about using others platform like github actions or circle ci.

**Comment by porunov** [comment]
@farodin91 I would like that `[full build]` would be finished for this PR, so that I know that it is consistent. After that we can omit `[full build]` in other PRs for now until we are switched to a better CI plan or another CI or split the tests on smaller parts. Other listed `[full build]` jobs for `0.3`, `0.4` and `master` branches are currently executed in my forked repository, so they don't take JanusGraph build time.

**Pull request merged**